### ISSUE TITLE
Have a single development server

### ISSUE THREAD

**Issue by DartBot** [dart-lang/pub#1182]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="96" height="96"hspace="10"></img></a> **Issue by [nex3](https://github.com/nex3)**
_Originally opened as dart-lang/sdk#21476_

---

Currently there are two servers necessary for full-fledged Dart development: the analysis server, which allows editors and other tools to access resolved information about the Dart code, and &quot;pub serve&quot;, which runs barback transformers and exposes web apps as an HTTP API.

A user should only have to start and maintain a single server per application. This server should provide both an interface for analysis and for serving assets over HTTP.

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

*Marked this as being blocked by dart-lang/sdk#21477.*

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

*Marked this as being blocked by dart-lang/sdk#20970.*

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

*Marked this as being blocked by dart-lang/sdk#21478.*

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

*Marked this as being blocked by dart-lang/sdk#21479.*

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

*Marked this as being blocked by dart-lang/sdk#17045.*
*Unmarked this as being blocked by dart-lang/sdk#21479.*

**Comment by DartBot** [comment]
<a href="https://github.com/seaneagan"><img src="https://avatars.githubusercontent.com/u/444270?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [seaneagan](https://github.com/seaneagan)**

----

> This server should provide both an interface for analysis and for serving assets over HTTP.

... and for dartdoc serving / editing?

https://github.com/sethladd/dartdoc

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

I think dartdoc is out-of-scope of the dev server, although hopefully it would make it easier to build a separate dartdoc server.

**Comment by DartBot** [comment]
<a href="https://github.com/seaneagan"><img src="https://avatars.githubusercontent.com/u/444270?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [seaneagan](https://github.com/seaneagan)**

----

Here is the bug about using pub serve for a dartdoc server:

https://github.com/sethladd/dartdoc/issues/2

In case you have any input on how that could work.

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

*Marked this as being blocked by dart-lang/sdk#22080, dart-lang/sdk#22081, dart-lang/sdk#22082.*

**Comment by matanlurey** [comment]
We are no longer supporting transformers or `pub serve/build` going forward (this tooling has moved elsewhere into the https://github.com/dart-lang/build repository and https://github.com/dart-lang/webdev CLI) so a fix for this is not planned. 

We'll be updating the tooling section of the Dart website in the coming weeks with more formal documentation on how to get started.
